src/BitStringAddresses/fockaddress.jl

```diff
@@ -1,15 +1,4 @@
-"""
-    AbstractFockAddress{N,M}
-
-Abstract type representing a Fock state with `N` particles and `M` modes.
-
-See also [`SingleComponentFockAddress`](@ref), [`CompositeFS`](@ref), [`BoseFS`](@ref),
-[`FermiFS`](@ref).
-"""
-abstract type AbstractFockAddress{N,M} end
-
-# `AbstractFockAddress`es can be reconstructed from their printout.
-Base.typeinfo_implicit(::Type{<:AbstractFockAddress}) = true
+# AbstractFockAddress is defined in Interfaces.jl, so we can use it here.
```

Comment:
If `AbstractFockAddress` is in Interfaces now, should we also move the basic function definitions (`num_particles` and such) and `SingleComponentFockAdress`?
Maybe since a user is unlikely to need to define new Fock address types, it's fine to leave it out of Interfaces?

Reply:
I've moved `num_particles`, `num_modes` and `num_components` but think I'd rather leave `SingleComponentFockAddress` where it is. If a user wants define a new subtype it doesn't matter that much which module it's in.